concourse/pipeline.yml

```diff
@@ -0,0 +1,114 @@
+---
+resources:
+  - name: fauna-js-repository
+    type: git
+    icon: github
+    source:
+      uri: https://github.com/fauna/faunadb-js
+      branch: oss-485-automate-js-driver-release-in-concourse-ci
+
+  - name: fauna-js-repository-docs
+    type: git
+    icon: github
+    source:
+      uri: git@github.com:fauna/faunadb-js.git
+      branch: gh-pages-test
+      private_key: ((fauna/repo.key))
+
+  - name: nodejs-image
+    type: registry-image
+    source:
+      repository: node
+      tag: 15.14.0-alpine3.10
+
+  - name: git-image
+      type: registry-image
+      source:
+        repository: alpine/git
+        tag: v2.30.2
+
+  - name: dind-image
+    type: registry-image
+    source:
+      repository: karlkfi/concourse-dcind
+
+jobs:
+  - name: integration
+    public: false
+    plan:
+      - get: fauna-js-repository
+      - get: dind-image
+      - task: integration-tests
+        image: dind-image
+        privileged: true
+        config:
+          platform: linux
+          inputs:
+            - name: fauna-js-repository
+          run:
+            path: entrypoint.sh
+            args:
+              - bash
+              - -ceux
+              - |
+                pwd
+                ls -la ./fauna-js-repository
+                # start containers
+                docker-compose -f fauna-js-repository/concourse/integration.yml run -e FAUNA_ROOT_KEY=((fauna.secret)) -e FAUNA_DOMAIN=((fauna.domain)) -e FAUNA_SCHEME=((fauna.scheme)) -e FAUNA_PORT=((fauna.port)) -e AUTH_0_URI=((fauna/auth0.uri)) -e AUTH_0_CLIENT_ID=((fauna/auth0/client.id)) -e AUTH_0_CLIENT_SECRET=((fauna/auth0/client.secret)) tests
+                # stop and remove containers
+                docker-compose -f fauna-js-repository/concourse/integration.yml down
+                # remove volumes
+                docker volume rm $(docker volume ls -q)
+  - name: publish
+    public: false
+    plan:
+      - get: nodejs-image
+      - get: fauna-js-repository
+        passed: [integration]
+        trigger: true
+      - task: browserify-and-npm-publish
+        image: nodejs-image
+        privileged: true
+        config:
+          platform: linux
+          inputs:
+            - name: fauna-js-repository
+          run:
+            path: ./fauna-js-repository/concourse/publish.sh
+  - name: publish-docs
+    public: false
+    plan:
+      - get: nodejs-image
+      - get: fauna-js-repository
+      - get: fauna-js-repository-docs
+        passed: [publish]
+        trigger: true
+      - task: generate-docs
+        image: nodejs-image
+        privileged: true
+        config:
+          platform: linux
+          inputs:
+            - name: fauna-js-repository
+          outputs:
+            - name: doc
+            - name: docs_version
+          run:
+            path: ./fauna-js-repository/concourse/generate_docs.sh
```

Comment:
I'm wondering here, if since we are making modifications to what is stored at the input for the fauna-js-repository, if those changes will be present to future tasks within this job.  If that is the case, would we be able to just do a commit after generating the docs, and then do a push and omit the `publish-docs-to-git-repo` task?

Reply:
Sure, I can re-work it this way.

I was looking at this example:
https://concourse-ci.org/nodejs-example.html
There are 2 tasks, the first one runs `npm install`, and then it sends the result as output to the following task that runs the tests. So I thought it is kind of a common way for concourse to work with the outputs this way.
But of course, it can be done in one task as well, can't say if it makes any difference.

Reply:
After merging these 2 tasks, I remembered why I did so:
I used 2 images, the first one is nodejs and the other one is git image, merging these 2 tasks together requires installing git if I use only nodejs image now.

Reply:
ah got it, I think its fine to keep it this way.

The other option would be similar to as you mentioned, we could create our own custom image that has the needed dependencies to do the publish.  I think this is a good starting point though and can always revisit if need be.